pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -251,6 +253,9 @@
     private volatile CompletableFuture<Void> closeFuture;
     // key is listener name , value is pulsar address and pulsar ssl address
     private Map<String, AdvertisedListener> advertisedListeners;
+    // For REST api, keep track of topics served by this pulsar.
+    @Getter
+    private ConcurrentOpenHashMap<String, ConcurrentOpenHashSet<Integer>> owningTopics = new ConcurrentOpenHashMap<>();
```

Comment:
Why do I need to cache all topics? We only need to determine whether the Topic is owned by current Broker, otherwise it will be redirected.
Broker already has Topic cache.
We can determine whether it belongs to the current cluster by checking the Namespace.

Reply:
I assume you're talking about the multiLayerTopicsMap in BrokerService, I took a look and it does contain topic ownership data, but it's not very handy in this use case which we need to keep track of serving topics as well as all partitions on a broker when publish to a partition topic so we can distribute message to all partitions.